File: solutions/clojure/raindrops/1/src/raindrops.clj
(ns raindrops
  (:require [clojure.string :refer [join]]))



(defn perhaps-pling [n]
  (if (zero? (mod n 3)) "Pling"  "")
  )

(defn perhaps-plang [n]
  (if (zero? (mod n 5)) "Plang"  "")
  )

(defn perhaps-plong [n]
  (if (zero? (mod n 7))"Plong"  "")
  )


(defn convert [number] ;; <- arglist goes here
      ;; your code goes here
  (let [rain 
        (join (str (perhaps-pling number) (perhaps-plang number) (perhaps-plong number)))] 
    (if (empty? rain) 
      (str number) rain))
)

File: solutions/julia/raindrops/1/raindrops.jl
function pling(number)
    if (number % 3 == 0) 
        "Pling"
    else
        ""
    end
end

function plang(number)
    if (number % 5 == 0) 
        "Plang"
    else
        ""
    end
end

function plong(number)
    if (number % 7 == 0) 
        "Plong"
    else
        ""
    end
end

function nodrops(number)
    if  !((number % 3 == 0) || (number % 5 == 0) || (number % 7 ==0)) 
        "$number"
    else
        ""
    end
end

function raindrops(number)
     string(pling(number), plang(number), plong(number), nodrops(number))
end
